services/bb-pacman/backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/SchemaService.java

```diff
@@ -48,13 +53,61 @@
     @Inject
     List<SchemaConfig> schemas;
 
+    @Inject
+    LedgerClient ledgerClient;
+
     // CRUD Methods
 
     public SchemaAPI addSchema(@NonNull String schemaId, @Nullable String label,
             @Nullable String defaultAttributeName) {
         return addSchema(schemaId, label, defaultAttributeName, false);
     }
 
+    public SchemaAPI createSchema(@NonNull String schemaName, @NonNull String schemaVersion,
+                        @NonNull List<String> attributes) {
+        SchemaAPI result = null;
+        try {
+            SchemaSendRequest request = SchemaSendRequest.builder()
+                    .schemaName(schemaName)
+                    .schemaVersion(schemaVersion)
+                    .attributes(attributes)
+                    .build();
+            Optional<SchemaSendResponse> response = ac.schemas(request);
+            if (response.isPresent()) {
+                // save it to the db...
+                SchemaSendResponse ssr = response.get();
+                result = this.addSchema(ssr.getSchemaId(), schemaName, null);
+            } else {
+                log.error("Schema not created.");
+            }
+
+            addCredentialDefinition(result);
+        } catch (IOException e) {
+            log.error("aca-py not reachable", e);
+        }
+        return result;
+    }
+
+    private void addCredentialDefinition(SchemaAPI schema) throws IOException {
+        if (schema != null) {
+            // for now hack this in, let's create a cred def with default values.
+            // no revocation!
+            CredentialDefinition.CredentialDefinitionRequest creddef = CredentialDefinition.CredentialDefinitionRequest.builder()
+                    .revocationRegistrySize(1000)
```

Comment:
Should this be set if revocation is not supported?

Reply:
we are not supporting revocation, that requires the ledger have a tails server which isn't in place yet.

Reply:
line 96 defines the revocationRegistrySize... wonder if we should scrap that attribute too?

Reply:
changed to 4 because that is the minimum, cannot set to zero.